src/StudyViewer/SingleStudyViewer.jsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+import { Space, Typography, Spin } from 'antd';
+import { FileOutlined, FilePdfOutlined } from '@ant-design/icons';
+import BackLink from '../components/BackLink';
+import { humanFileSize } from '../utils.js';
+import { ReduxStudyDetails, fetchDataset, fetchFiles } from './reduxer';
+import getReduxStore from '../reduxStore';
+import './StudyViewer.css';
+
+const { Title } = Typography;
+
+class SingleStudyViewer extends React.Component {
+  componentDidMount() {
+    getReduxStore().then(
+      store =>
+        Promise.all(
+          [
+            store.dispatch(fetchDataset(decodeURIComponent(this.props.match.params[0]))),
+            store.dispatch(fetchFiles('object', decodeURIComponent(this.props.match.params[0]))),
+            store.dispatch(fetchFiles('open-access', decodeURIComponent(this.props.match.params[0]))),
+          ],
+        ));
+  }
+
+  render() {
+    if (!this.props.dataset || this.props.dataset.length === 0) {
+      return (
+        <div className='study-viewer'>
+          <div className='study-viewer_loading'>
+            <Spin size='large' tip='Loading data...' />
+          </div>
+        </div>
+      );
+    }
+    const dataset = this.props.dataset[0];
```

Comment:
would it make sense to feed `SingleStudyViewer` a single dataset instead of a list, or does it make the code more complicated?

Reply:
thought about it and i think it would make the code more complex since it comes from redux state

Reply:
actually, back to this, turns out I do need to separate them because otherwise it would not be possible to trigger the refetching since now we need to use `getDerivedStateFromProps` to get data type and row accessor value from URL